spec/services/pii/re_encryptor_spec.rb

```diff
@@ -2,32 +2,60 @@
 
 RSpec.describe Pii::ReEncryptor do
   describe '#perform' do
-    it 're-encrypts PII using new code' do
-      profile = create(:profile, :active, :verified, pii: { ssn: '1234' })
-      user = profile.user
-      user_session = {
-        decrypted_pii: { ssn: '1234' }.to_json,
-      }
+    let(:active_ssn) { '1234' }
+    let(:active_pii) { { ssn: active_ssn } }
 
-      pii_attributes = Pii::Attributes.new
-      pii_attributes[:ssn] = '1234'
+    let(:pending_ssn) { '5678' }
+    let(:pending_pii) { { ssn: pending_ssn } }
 
-      expect(user.active_profile).to receive(:encrypt_recovery_pii).
-        with(pii_attributes).and_call_original
-      expect(user.active_profile).to receive(:save!).and_call_original
+    let(:user) { create(:user) }
+    let(:user_session) { {}.with_indifferent_access }
+    let(:pii_cacher) { Pii::Cacher.new(user, user_session) }
+    let(:re_encryptor) { Pii::ReEncryptor.new(user: user, user_session: user_session) }
 
-      Pii::ReEncryptor.new(user: user, user_session: user_session).perform
+    context 'when the user has an active profile and a pending profile' do
+      let(:expected_pii) do
+        expected_pii = Pii::Attributes.new
+        expected_pii[:ssn] = active_ssn
+        expected_pii
+      end
```

Comment:
Having this as a separate let in each context isn't adding anything, since there's only one spec in each context. I think it would be clearer to have expected_pii initialized before it's used, the way it used to be, moving lines 18 and 19 to just above line 36.

Reply:
Moved.